### ISSUE TITLE
Upgraded from version 0.7.5 to latest develop and facing issues

### ISSUE THREAD

**Issue by sushovan23** [llazzaro/django-scheduler#283]
Hello,
I had developed UI using old version of django-scheduler 0.7.5
But during my testing I found the issues with get occurences list. After checking the commit history I saw there was a recent fix so I took latest code and merged with my UI.
I did all migrations and updated the call to scheduletags also.
My daily view is not showing currently.

Here is the use case:
Created a daily recurrence event for 7 days from 10am to next day 10am.
` start               | end                 | title           | description | created_on          | updated_on          | end_recurring_period | calendar_id | creator_id | rule_id
---------------------+---------------------+-----------------+-------------+---------------------+---------------------+----------------------+-------------+------------+---------
 2017-01-13 15:00:00 | 2017-01-14 15:00:00 | smajum002c,test |             | 2017-01-24 01:14:44 | 2017-01-24 01:14:44 | 2017-01-20 05:00:00  |          11 |          1 |      10` 

rule used is:
` 10 | DAILY   | Occurs every day   | DAILY     | NULL`

I'm displaying the daily view like:
{% daily_table period 0 24 60 %}

However instead of entire day showing as occupied it's only showing the events at slot for 9 AM and 10 AM EST and also for 9PM and 10 PM EST.

Earlier it used to show entire day as blocked or filled and it was a very useful view to check daily events.

Can someone please tell me how to display daily view for complete set currently? Any input would be greatly appreciated?

**Comment by mpaolino** [comment]
@sushovan23 I've tried your example on my latest ``develop`` branch and it works fine. Please notice that there where several other merged fixes after #271.

Can you please share what is the HEAD of the develop branch you're testing with? Also, are you using ```USE_TZ = True``` in your configs? If so, what timezone have you configured under ``TIME_ZONE`` (if any)?

**Comment by sushovan23** [comment]
@mpaolino Thanks for checking this. Yes I'm using latest develop at commit version  35b0654.
It has only change for quotes #281 after your fixes. 
Yes I'm using `USE_TZ = True` and `TIME_ZONE = 'America/New_York'`

Let me know how are you displaying the daily calendar. I can also take from your develop branch or any version here

**Comment by mpaolino** [comment]
I couldn't reproduce your bug but I found another issue, the end_recurring_period is not being honored, and the occurrences in weekly views from your test case never ends. I'm making a fix for this, lets try to take it from there as I don't want to tackle 2 issues at once. Once this is fixed I'll point you to my branch if you want to try it.

**Comment by sushovan23** [comment]
@mpaolino I missed taking some changes but now I have everything and I also found same issue which you pointed out regarding end occurrences not respected in daily and weekly view. Monthly view is good but daily and weekly view doesn't end the occurrences.

My original issue related to missing slots doesn't occur any more. Glad we both are on same page now. I will test out your fix immediately once it's ready.

**Comment by mpaolino** [comment]
@sushovan23 do you mind testing my branch under https://github.com/mpaolino/django-scheduler (``develop`` branch)?. I'll make a PR if you confirm its fixed for you.

**Comment by sushovan23** [comment]
@mpaolino : I took your latest change in events.py. I can confirm that now it fixes the daily view , weekly view and monthly view.

**Comment by mpaolino** [comment]
@sushovan23 thanks, I've just pushed another fix on my branch for events that don't have an ``end_recurring_period``. I'll make a PR, feel free to pull it from my branch or wait untils its merged.

**Comment by sushovan23** [comment]
@mpaolino I took that change

**Comment by llazzaro** [comment]
@sushovan23 @mpaolino I will merge the PR soon. thanks!

**Comment by sushovan23** [comment]
@mpaolino : Found another issue while editing occurrences. I am not getting any occurrences back from get_occurrence. get_occurrence is called from edit occurrence url mapping.
get_occurrence inturn calls ```
event.get_occurrence(
            datetime.datetime(int(year), int(month), int(day), int(hour),
                              int(minute), int(second), tzinfo=tzinfo))
```
this method return none as occurrence.
Does that function also needs to be changed? Could you let me know if I'm missing something here?

**Comment by mpaolino** [comment]
@sushovan23 I'm learning on the go, let me check it out.

**Comment by mpaolino** [comment]
@sushovan23 please create another ticket with this new issue, and provide test data, it would be easy to track.

**Comment by sushovan23** [comment]
@mpaolino : Sure here is the test data:
Created one event 24 hrs duration with daily re-occurrence for a week. Then from daily calendar view and i'm clicking one any occurrence and clicking the option to edit 'This' occurrence from the popup dialog,

data in database:
```
+-----+---------------------+---------------------+-----------------+-------------+---------------------+---------------------+----------------------+-------------+------------+---------
| id  | start               | end                 | title           | description | created_on          | updated_on          | end_recurring_period | calendar_id | creator_id | rule_id
+-----+---------------------+---------------------+-----------------+-------------+---------------------+---------------------+----------------------+-------------+------------+---------
| 206 | 2017-01-13 15:00:00 | 2017-01-14 15:00:00 | smajum002c,test |             | 2017-01-24 14:22:02 | 2017-01-24 14:22:02 | 2017-01-20 05:00:00  |          11 |          1 |      10
```


Then it should go to edit occurrence page but it is failing since get_occurrence() is views.py has an issue in getting occurrence from method event.getoccurence() which returns None. Thus it doesn't go to edit occurrence page anymore.
